xoinvader/background.py

```diff
@@ -0,0 +1,261 @@
+
+    def _fill(self):
+        """Fills the whole background"""
+        self.background = []
+        for i in range(self._h):
+            self.background.append(self._advance_chunk(1))
+
+
+    def start(self, filled=False):
+        """Start background from the beginning. If `filled` parameter is
+        provided, fills the background starting with first chunk.
+
+        :param filled: if true, perform initial background fill
+        :type filled: boolean"""
+        self._ticks_since_last_update = 0
+        self._current_chunk_num = 0
+        self._current_chunk = self.chunks[0]
+        self._chunk_line = 0
+        if filled:
+            self._fill()
+        else:
+            self.clear()
+        self.update_surface()
+
+    def _advance_chunk(self, advance):
+        """Return current chunk line and update current chunk's position,
+        or current chunk if there's no lines left in it.
+
+        :return: current line in chunk sequence
+        :rtype: string
+        """
+        if not self._current_chunk:
+            return ' ' * self._w
+
+        # start checks
+        if (self._chunk_line < self._current_chunk.length and
+                self._chunk_line >= 0):
+            l = self._current_chunk[self._chunk_line]
+            self._chunk_line += advance
+            return l
+
+
+        # chunk ended! now we decide what to do
+        if self.loop:
+            if self._chunk_line < 0:
+                self._chunk_line = self._current_chunk.length - 1
+            else:
+                self._chunk_line = 0
+            l = self._current_chunk[self._chunk_line]
+            self._chunk_line += advance
+            return l
+
+        # next chunk or stop advancing
+        self._current_chunk_num += advance
+        if (self._current_chunk_num < len(self.chunks) and
+                self._current_chunk_num >= 0):
+            # okay, just set new chunk
+            self._current_chunk = self.chunks[self._current_chunk_num]
+            self._chunk_line = 0
+            l = self._current_chunk[self._chunk_line]
+            self._chunk_line += advance
+            return l
+
+        if self.loop_all:
+            if self._current_chunk_num < 0:
+                self._current_chunk_num = len(self.chunks) - 1
+                self._current_chunk = self.chunks[-1]
+                self._chunk_line = self._current_chunk.length - 1
+            else:
+                self._current_chunk_num = 0
+                self._current_chunk = self.chunks[0]
+                self._chunk_line = 0
+            l = self._current_chunk[self._chunk_line]
+            self._chunk_line += advance
+            return l
+
+        # looks like we don't have any more chunks, and we're not looping
+        # thus return void line
+        self._current_chunk = None
+        return ' ' * self._w
+
+    def get_render_data(self):
+        return [Point(0, 0)], self._background_surface.get_image()
+
+    def update_surface(self):
```

Comment:
Yes. We provide direct access to `background` filed, and as outside code may change the background list, it also should have a way to regenerate background immediately, not waiting for the next time it updates via own `update` method.